packages/node/src/core/evm/providers.ts

```diff
@@ -0,0 +1,113 @@
+import { ethers } from 'ethers';
+import { Airnode, Convenience } from './contracts';
+import { go, retryOperation } from '../utils/promise-utils';
+import * as logger from '../logger';
+import * as wallet from './wallet';
+import { LogsData } from '../../types';
+
+interface BaseFetchOptions {
+  airnodeAddress: string;
+  convenienceAddress: string;
+  provider: ethers.providers.JsonRpcProvider;
+}
+
+interface FindOptions extends BaseFetchOptions {
+  providerId: string;
+}
+
+interface CreateOptions extends BaseFetchOptions {
+  adminAddress: string;
+  airnodeAddress: string;
+  provider: ethers.providers.JsonRpcProvider;
+  xpub: string;
+}
+
+interface ProviderWithBlockNumber {
+  adminAddress: string;
+  blockNumber: number;
+  providerExists: boolean;
+  xpub: string;
+}
+
+export async function findWithBlock(fetchOptions: FindOptions): Promise<LogsData<ProviderWithBlockNumber | null>> {
+  const convenience = new ethers.Contract(fetchOptions.convenienceAddress, Convenience.ABI, fetchOptions.provider);
+  const contractCall = () => convenience.getProviderAndBlockNumber(fetchOptions.providerId);
+  const retryableContractCall = retryOperation(2, contractCall, { timeouts: [4000, 4000] }) as Promise<any>;
+
+  const fetchLog = logger.pend('INFO', 'Fetching current block and provider admin details...');
+
+  const [err, res] = await go(retryableContractCall);
+  if (err || !res) {
+    const errLog = logger.pend('ERROR', 'Unable to fetch current block and provider admin details', err);
+    return [[fetchLog, errLog], null];
+  }
+
+  const data: ProviderWithBlockNumber = {
+    adminAddress: res.admin,
+    // Converting this BigNumber to a JS number should not throw as the current block number
+    // should always be a valid number
+    blockNumber: res.blockNumber.toNumber(),
+    providerExists: !!res.xpub && res.xpub !== '',
+    xpub: res.xpub,
+  };
+
+  const blockLog = logger.pend('INFO', `Current block: ${res.blockNumber}`);
+
+  if (!data.providerExists) {
+    const providerLog = logger.pend('INFO', 'EVM provider not found');
```

Comment:
The "provider" in the protocol is short for "API provider", though I never use it as such.  I think this "EVM provider" is referring to the same provider, so it's not accurate.

Reply:
I didn't have any problems with this implementing the protocol because I didn't deal with EVM providers much but obviously it's pretty confusing. I'm open to suggestions.

Reply:
I don't have any better suggestions at the moment